items.py

```diff
@@ -34,87 +33,62 @@ def create_request_objects(item, property1, property2, quantity):
 
-    items = []
+    props = properties[item_type]
 
-    for item_object in filter(lambda x: not re.search("DMG", x.text), soup.findAll('div', {"class": 'weapon'})):
-        # get nonempty lines
-        lines = list(filter(None, item_object.text.splitlines()))
-
-        # get name
-        name = re.sub("\d+: ", "", lines[0])
-
-        # get type and rarity
-        try:
-            type_name, rarity = lines[1].split(",")[0:2] if lines[1].split(",").__len__() > 1\
-                else (lines[1].split(",")[0], "common")
-        except Exception as ex:
-            print(ex)
-            print(lines[1].split(","))
-            print(item_object.text)
-            continue
-
-        # there will be a space after the comma, at the beginning of rarity, as well as a (requires Attunement) after it
-        # so just extract rarity
-        rarity, attunement = rarity.split(" ")[1::2] if rarity is not "common" else (rarity, "")
-        attunement = "TRUE" if re.search("Attunement", attunement, re.IGNORECASE) else ""
-
-        # get effect(s)
-        # replace with break so it works in html
-        effect = "<br>".join(lines[2:])
-
-        # determine slot
-        if re.match("Armor", type_name):
-            slot = "Armor"
-        elif re.match("Weapon", type_name):
-            slot = "Weapon"
-        else:
-            if re.search('Helm|Cap|Hat|Circlet|Mask|Tiara', name, re.IGNORECASE):
-                slot = "Head"
-            elif re.search('Gauntlets|Bracers', name, re.IGNORECASE):
-                slot = "Wrist"
-            elif re.search('Boots', name, re.IGNORECASE):
-                slot = "Feet"
-            elif re.search('Collar|Necklace|Amulet|Pendant|Medallion', name, re.IGNORECASE):
-                slot = "Neck"
-            else:
-                expression = re.compile('(Belt|Ring|Cloak|Gloves)')
-                match = expression.search(name)
-                slot = match.group(1) if match else ""
-
-        items.append({
+    if not props:
+        return []
+
+    name = ""
+    effect = ""
+
+    if property1 and property1 == "weak":
+        prop = random.choice(props)
+        name = prop["prop1"]
+        effect += prop["effect"].strip()
+
+    item_ops = random.choice(ops)
+
+    item = random.choice(item_ops) if type(item_ops) == list else item_ops
+
+    typ = "{}{}".format(item_type, "({})".format(item_ops[0]) if type(item_ops) == list else "")
+
+    name += " " + item if name != "" else item
+
+    if property2 and property2 == "weak":
+        prop = random.choice(props)
+        name += " " + prop["prop2"]
+        effect += "{}{}".format("\n" if effect != "" else "", prop["effect"].strip())
+
+    return {
             "name": name,
-            "type": type_name,
-            "rarity": rarity,
-            "attunement": attunement,
-            "slot": slot,
+            "type": typ,
+            "rarity": "Uncommon",
+            "attunement": "TRUE",
+            "slot": item_type,
```

Comment:
This will be Wonderous item, but should be Head, Belt, etc. old code worked, pull over